src/postgres-util/src/tunnel.rs

```diff
@@ -112,15 +114,34 @@ pub const DEFAULT_SNAPSHOT_STATEMENT_TIMEOUT: Duration = Duration::ZERO;
 ///
 /// This wraps [`tokio_postgres::Config`] to allow the configuration of a
 /// tunnel via a [`TunnelConfig`].
-#[derive(Debug, PartialEq, Clone)]
+#[derive(Debug, Clone)]
 pub struct Config {
     inner: tokio_postgres::Config,
     tunnel: TunnelConfig,
+    secrets_reader: Arc<dyn SecretsReader>,
+    ssh_tunnel_manager: SshTunnelManager,
+}
+
+impl PartialEq<Config> for Config {
+    fn eq(&self, other: &Config) -> bool {
+        self.inner == other.inner && self.tunnel == other.tunnel
+    }
```

Comment:
`Config` structs are meant to be plain old data. We need to find a different API here that preserves that property. I think there's a way of slicing this so that you don't need the `SecretsReader` and the `SshTunnelManager` is instead passed into to `connect`.